rust/rbac-registration/src/cardano/cip509/types/role_data_record.rs

```diff
@@ -0,0 +1,129 @@
+//! Role data record where each role data fields are stored.
+
+use std::collections::HashMap;
+
+use c509_certificate::c509::C509;
+use ed25519_dalek::VerifyingKey;
+use pallas::ledger::addresses::ShelleyAddress;
+use x509_cert::Certificate as X509;
+
+use super::{PointData, PointTxnIdx};
+use crate::cardano::cip509::extract_key;
+
+/// Actual data of key local ref. Containing X509, C509 and public key.
+#[derive(Debug, Clone)]
+pub enum CertOrPk {
+    /// X509 certificate, None if deleted.
+    X509(Option<Box<X509>>),
```

Comment:
I think it makes sense to provide our own type instead of using Option here.
I am not sure of the naming, but it would eliminate a potential confusion with `Option`, so the type itself will be self documented
```
enum Reg<T> {
  Reg(Box<T>),
  Deleted,
}

Reply:
I think `Option`  is simpler.
In this context there is just 2 state, having certificate (Some cert) or certificate is deleted (so this is None)